File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        if(s.length() < k) return false;
        if(s.length() == k) return true;

        int[] cnt = new int[26] ;
        for(char i : s.toCharArray()){
            cnt[i - 'a'] ++ ;
        }

        int odd = 0 ;
        for(int i = 0 ; i < 26 ; i ++)
            odd += cnt[i] % 2 ;

        return odd <= k ;
    }
}

File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) < k:
            return False 
        
        cnt = [0] * 26 
        for i in s:
            cnt[ord(i) - ord('a')] += 1
        
        odd = 0
        for i in range(26):
            odd += cnt[i] % 2 
        
        return odd <= k